core/src/main/java/org/elasticsearch/indices/cache/query/terms/TermsLookup.java

```diff
@@ -125,6 +125,8 @@ public static TermsLookup parseTermsLookup(XContentParser parser) throws IOExcep
                     throw new ParsingException(parser.getTokenLocation(), "[terms] query does not support [" + currentFieldName
                             + "] within lookup element");
                 }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(), "[terms] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?